website/common/script/ops/buy/buyMarketGear.js

```diff
@@ -24,6 +24,10 @@ export class BuyMarketGearOperation extends AbstractGoldItemOperation {
     return false;
   }
 
+  canUserPurchase (user, item) {
+    return super.canUserPurchase(user, item) && (item.klass !== 'special' && item.klass !== user.stats.class);
```

Comment:
`canUserPurchase` doesn't return a bool, I guess the name is misleading, sorry about that!

`super.canUserPurchase` throws errors in the conditions if the user can't purchase it, so your prior code was right, you only needed to move it here under the the `super.canUserPurchase`-call